frontend/lib/messages/view/conversation_detail.view.dart

```diff
@@ -0,0 +1,185 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:utm_marketplace/messages/components/message_bubble.component.dart';
+import 'package:utm_marketplace/messages/view_models/message.viewmodel.dart';
+import 'package:utm_marketplace/shared/components/loading.component.dart';
+import 'package:go_router/go_router.dart';
+
+
+class ConversationDetailView extends StatefulWidget {
+  final String conversationId;
+
+  const ConversationDetailView({
+    super.key,
+    required this.conversationId,
+  });
+
+  @override
+  State<ConversationDetailView> createState() => _ConversationDetailViewState();
+}
+
+class _ConversationDetailViewState extends State<ConversationDetailView> {
+  late MessageViewModel viewModel;
+  final TextEditingController _messageController = TextEditingController();
+  final ScrollController _scrollController = ScrollController();
+
+  @override
+  void initState() {
+    super.initState();
+    viewModel = Provider.of<MessageViewModel>(context, listen: false);
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      viewModel.fetchConversation(widget.conversationId);
+    });
+  }
+
+  @override
+  void dispose() {
+    _messageController.dispose();
+    _scrollController.dispose();
+    super.dispose();
+  }
+
+  void _scrollToBottom() {
+    if (_scrollController.hasClients) {
+      _scrollController.animateTo(
+        _scrollController.position.maxScrollExtent,
+        duration: const Duration(milliseconds: 300),
+        curve: Curves.easeOut,
+      );
+    }
+  }
+
+  void _sendMessage() async {
+    if (_messageController.text.trim().isEmpty) return;
+    
+    viewModel.messageText = _messageController.text;
+    final success = await viewModel.sendMessage();
+    
+    if (success) {
+      _messageController.clear();
+      _scrollToBottom();
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        leading: IconButton(
+          icon: const Icon(Icons.arrow_back),
+          onPressed: () => context.pop(),
+          color: const Color(0xFF2F9CCF),
+        ),
+        title: Consumer<MessageViewModel>(
+          builder: (_, messageViewModel, __) {
+            final conversation = messageViewModel.currentConversation;
+            if (conversation == null) return const Text('Loading...');
```

Comment:
Spend some time inspecting this logic. Right now, when you switch between different conversation detail views, there's a period (before the current conversation is loaded) where the app bar still shows the previous conversation header. If possible, we want to remove this.

I suspect this is caused by the fact that we still store the previous conversation in the ViewModel. Ideally, I think each Conversation will be cached inside the ViewModel and you simply load them when switching between conversation views. Note: this will also require you to implement paging (since messages might become very long).

Reply:
For a bit of background, the ChangeNotifierProvider lazy creates its ViewModel but that ViewModel instance is now persistent in the application. That's why ViewModel states are persisted even after navigating to and from pages. This is actually good as caching messages is definitely something we want (imagine having to wait for Whatsapp to load your most recent 100 messages each time you switch between chats).

Reply:
Again, I think leave this out for later once the backend is more refined.